src/cpuinfo/SDL_cpuinfo.c

```diff
@@ -840,6 +840,7 @@ static const char *SDL_GetCPUName(void)
 int SDL_GetCPUCacheLineSize(void)
 {
     const char *cpuType = SDL_GetCPUType();
+    int failsafe_cacheline_size;
```

Comment:
move this into `#ifdef __LINUX__` block where it is used

Reply:
I leaved it at the top because if we want to add more platform lever queries on else case like for checking on ```__WIN32__``` we can use that variable and return it so we won't have to declare one for each of them. I can just push a commit to move it into ```#ifdef __LINUX__``` if you disagree.

Reply:
It will lead to `-Werror=unused` warnings..

Reply:
Okay I will just push a commit then.

Reply:
Another thing we might do it you approve we can modify return statement so that we first write into a local variable called cache_size and return it after, with that we won't have to add another local variable for fail safe and simply write into that. With that we won't have any unused warnings/failures either.

Reply:
That seems like a good idea. Please squash these changes into a single commit for review when you're done.

Reply:
Sure! Going to test this on FreeBSD as well, as soon as it's done going to push a commit for it.